db/migrate/20211202071418_environment_in_taxonomy_ignore_types.rb

```diff
@@ -0,0 +1,16 @@
+class EnvironmentInTaxonomyIgnoreTypes < ActiveRecord::Migration[6.0]
+  def up
+    organizations = Organization.where("ignore_types LIKE '%Environment%'")
+    locations = Location.where("ignore_types LIKE '%Environment%'")
+
+    (organizations + locations).each do |tax|
+      if Foreman::Plugin.find(:foreman_puppet)
+        new_types = tax.ignore_types.map { |type| type == "Environment" ? "ForemanPuppet::Environment" : type }
```

Comment:
This is little problematic, I believe we should do this migration in `foreman_puppet` as we will have to deal with:

https://github.com/theforeman/foreman/blob/b1f03b375e18c8da498ba0944b2565f53441833c/app/services/tax_host.rb#L32

which will cause `ids['foreman_puppet/environment_ids']` and that is meaningless.
above that, now we do not have `environment_ids` in `TaxHost::HASH_KEYS`